bitlum.go

```diff
@@ -0,0 +1,763 @@
+  			}
+		}
+	`
+
+	req.Variables = depositRequestVariables{
+		Assets: []string{asset},
+		Offset: offset,
+		Limit:  limit,
+	}
+
+	resp := struct {
+		responseBase
+		Data struct {
+			Deposits []Deposit `json:"balanceUpdateRecords"`
+		}
+	}{}
+
+	respJSON, err := c.do(req)
+	if err != nil {
+		return nil, errors.New("failed to do request: " + err.Error())
+	}
+
+	if err := json.Unmarshal(respJSON, &resp); err != nil {
+		return nil, errors.New("failed to json.Unmarshal resp: " +
+			err.Error())
+	}
+
+	if err := resp.Error(); err != nil {
+		return nil, errors.New("bitlum server-side error: " +
+			err.Error())
+	}
+
+	return resp.Data.Deposits, nil
+}
+
+// Order is an exchange order to buy or sell stock. Market contains
+// two currencies: left one is money and right - stock. For example
+// Market{BTC,LTC} means that BTC is a money and LTC - stock.
+type Order struct {
+	// ID is a exchange specific order ID
+	ID int64
+
+	// Status is order status: pending, finished or canceled
+	Status string
+
+	// Either amount of money or stock depending on direction
+	// of order: buy or sell. Now only buy direction used so this is
+	// always should be stock amount.
+	Amount decimal.Decimal
+
+	// Price of 1 stock in money currency.
+	Price decimal.Decimal
+
+	// DealMoney is the amount of money which were involved in the
+	// order.
+	DealMoney decimal.Decimal
+
+	// DealStock is the amount of stock which were involved in the
+	// order.
+	DealStock decimal.Decimal
+
+	// Left is the amount of funds left in the market without being
+	// handled.
+	Left decimal.Decimal
+}
+
+// orderRequestVariables is a query variables used in request
+// in bitlum Order method.
+type orderRequestVariables struct {
+	ID int64 `json:"id"`
+}
+
+// Order returns order with specified id
+func (c *Client) Order(id int64) (Order, error) {
+
+	var req request
+
+	req.Query = `
+		query GetOrder($id: Int!) {
+  			order(id: $id) {
+				id
+    			status
+				dealStock
+				dealMoney
+				amount
+				price
+  			}
+		}
+	`
+
+	req.Variables = orderRequestVariables{id}
+
+	resp := struct {
+		responseBase
+		Data struct {
+			Order Order
+		}
+	}{}
+
+	respJSON, err := c.do(req)
+	if err != nil {
+		return Order{}, errors.New("failed to do request: " + err.
+			Error())
+	}
+
+	if err := json.Unmarshal(respJSON, &resp); err != nil {
+		return Order{}, errors.New("failed to json.Unmarshal resp: " +
+			err.Error())
+	}
+
+	if err := resp.Error(); err != nil {
+		return Order{}, errors.New("bitlum server-side error: " +
+			err.Error())
+	}
+
+	return resp.Data.Order, nil
+}
+
+// createOrderRequestVariables is a query variables used in request
+// in bitlum CreateOrder method.
+type createOrderRequestVariables struct {
+	Market string          `json:"market"`
+	Amount decimal.Decimal `json:"amount"`
+}
+
+// CreateBidOrder creates bid order on market. Bid order means that
+// market[0] asset is used to buy market[1] asset
+func (c *Client) CreateOrder(market string,
```

Comment:
Please rephrase the comment, would be better to get explanation from investopedia.